pkgs/swift2objc/lib/src/parser/parsers/declaration_parsers/parse_function_declaration.dart

```diff
@@ -91,6 +91,22 @@ ParsedFunctionInfo parseFunctionInfo(
 
   final openParen = tokens.indexWhere((tok) => matchFragment(tok, 'text', '('));
   if (openParen == -1) throw malformedInitializerException;
+
+  final prefixAnnotations = <String>{};
+  while (true) {
+    final tok = tokens.first['kind'].get<String>();
+    if (tok == 'keyword') {
```

Comment:
If this check is false, you'll be stuck in an infinite loop, because tokens are only being consumed by `maybeConsume` in the true case. So you'll just keep checking the first token over and over (this is probably why the tests are timing out). So this outer if statement needs an else case. If there's no valid token stream that can have a non-keyword before `func`, then just `throw malformedInitializerException` in the else case.